internal/aws/containerinsight/utils.go

```diff
@@ -245,3 +246,10 @@ func initMetric(ilm pmetric.ScopeMetrics, name, unit string) pmetric.Metric {
 
 	return metric
 }
+
+func IsHostProcessContainer() bool {
```

Comment:
does this need to be duplicated in both the agent and contrib?

Let's have one source of truth.  Either put the check here and have the agent call this code OR have the agent do the check and pass the context in to the container insights plugin via config.

The config approach is probably the right approach

Reply:
During discussion on meeting, We are keeping this duplicate code since this code will be removed once EKS Windows AMIs upgrades containerd to 1.7 which is planned for early this year(Q1/Q2). Added comment in code to remove these temporary workarounds and cut ticket to EKS Windows team to remove these once upgraded AMIs are out.